internal/state/execution.go

```diff
@@ -606,14 +580,19 @@ func fireEvents(
 		}
 	}
 
-	for i, tx := range block.Data.Txs {
-		if err := eventBus.PublishEventTx(ctx, types.EventDataTx{TxResult: abci.TxResult{
-			Height: block.Height,
-			Index:  uint32(i),
-			Tx:     tx,
-			Result: *(abciResponses.DeliverTxs[i]),
-		}}); err != nil {
-			logger.Error("failed publishing event TX", "err", err)
+	//TODO strengthen this condition
+	if len(abciResponses.FinalizeBlock.Txs) != 0 {
```

Comment:
I'm not actually sure I understand what this check exists to do. If we're worried that the length of `FinalizeBlock.Txs` is not the same as `Data.Txs`, we should check that explicitly and error.

Reply:
I had a similar doubt when I rebased this code, hence I added the TODO... unfortunately, I then forgot to tackle it.
Thanks for pointing this out. I'll handle it in this PR